### ISSUE TITLE
BG1: Screens are exited as soon as the character is on the marker level

### ISSUE THREAD

**Issue by bsdcode** [gemrb/gemrb#2014]
#### Bug description
Characters switch screens as soon as they are almost on the same level as the marker. They do not go the whole way. Example is the second floor of the Candlekeep Inn. If you select to go downstairs then the character will not go the whole way around the stairs and not actually step down the stairs. Instead it will switch to the first floor as soon as it has passed the marker. The screenshot shows the marker at the stairs and the mousepointer shows roughly the position at which the character will switch screens to the first floor. It will not go around the stair railings.

#### Expected behavior
Character goes the whole way to the marker before switching screens.

#### Screenshots
![bg1_gemrb](https://github.com/gemrb/gemrb/assets/155195419/46164e08-4262-49fd-8f95-d1b0035c5f9e)

GemRB version (check as many as you know apply)
- [x] master as of this issue
- [ ] 0.9.2
- [ ] 0.9.1
- [ ] 0.9.0

Video Driver (check as many as you know apply)
- [ ] SDL1.2
- [x] SDL2 built with `OPENGL_BACKEND` enabled
- [ ] SDL2 without `OPENGL_BACKEND` enabled

**Comment by burner1024** [comment]
I think the originals don't go the whole way necessarily, but they do go closer than this.

**Comment by bsdcode** [comment]
Yes, the originals go halfway down the stairs before the area is changed. The green marker in the screenshot is the position at which the area is changed.

![bg1_wine](https://github.com/gemrb/gemrb/assets/155195419/a077ab23-1a25-4cb9-8179-0407ba396325)

**Comment by lynxlynxlynx** [comment]
It could be several of our constants for good-enough that's causing us to stop pathing. MAX_OPERATING_DISTANCE, the cutoff in the pathfinder, something else in the actual movement logic ... For example, perhaps it required a clear direct path, but that could then get us stuck if not implemented correctly. Lowering these constants needs to be done with care, since not only it can make things non-interactable, but also since larger values are cosier for players.
